### PR TITLE
Share structural variants as single genes to the MatchMaker

### LINKED ISSUE #1583: Submit genes to MME when variant is SV variant
Allow submission of SNVs only

### PR DESCRIPTION
**Pull request by northwestwitch**
Scout currently supports only sharing of SNVs to MatchMaker. And you can only submit all pinned variants for a case. Fix #1583

This PR allows to to:

- Share max 3 variants (SNVs or **SVs**) to MatchMaker Exchange
- Allows user to choose which variants should be shared from the list of pinned variants:
image

**How to test**:
1. Locally: uncomment the following lines in config file:
```
# MME_ACCEPTS = "application/vnd.ga4gh.matchmaker.v1.0+json"
# MME_URL = "http://localhost:9020"
# MME_TOKEN = "DEMO"
```
2. On another terminal, install and run [patientMatcher](https://github.com/Clinical-Genomics/patientMatcher) on port 9020:
`pmatcher run -p 9020`

3. Grant Scout demo user permissions to share variants to the matchmaker:
 `scout --demo update user -u clark.kent@mail.com -r mme_submitter`

4. Launch Scout and pin variants (SNVs and SVs). Go to case page and submit variants of choice to MME
**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/cases/templates/cases/utils.html [hunk 1]
```diff
@@ -67,24 +67,38 @@ <h5 class="modal-title" id="exampleModalLabel">MatchMaker Exchange patient submi
...
-                    <div class="d-flex justify-content-between">
-                      <div data-placement="top" title="genomicfeatures">
-                        Share pinned variants details
-                      </div>
-                      <div>
-                        <input type="radio" class="ios8-switch" id="genomicfeatures" name="genomicfeatures" value="variants" checked>
-                        <label for="genomicfeatures"></label>
-                      </div>
+                    Select <u><b>max 3</b></u> variants/genes from the pinned variants list (only gene names can be shared for SVs)<br>
+                    <div style="max-height:200px; overflow-y: scroll;"">
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/utils.html:71]: 💯

File: scout/server/blueprints/cases/templates/cases/utils.html [hunk 2]
```diff
@@ -67,24 +67,38 @@ <h5 class="modal-title" id="exampleModalLabel">MatchMaker Exchange patient submi
...
+                      {% for var in suspects %}
+                        {% if var.category == "snv" %} <!-- Share either gene names or variant details for SNVs -->
+                          <input type="checkbox" name="selected_var" id="var_details" value="{{var._id}}">{{ pretty_variant(var) }}<span class="badge badge-success">SNV</span><br>
+                        {% elif var.category == "sv" %} <!-- Share only gene names from SVs -->
+                          {% for gene in var.genes %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/utils.html:76]: 👍

File: scout/server/blueprints/cases/templates/cases/utils.html [hunk 3]
```diff
@@ -268,7 +282,33 @@ <h5 class="modal-title" id="exampleModalLabel">Beacon submission</h5>
   {% if n_primary_txs.count > 5 %}
   .. other transcripts available for this variant are not shown.<br><br>
   {% endif %}
+{% endmacro %}
+
+{% macro pretty_variant(variant) %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/utils.html:287]: 👍

File: scout/server/config.py [hunk 4]
```diff
@@ -34,7 +34,7 @@
 # - Tested with PatientMatcher (https://github.com/Clinical-Genomics/patientMatcher) -
 # MME_ACCEPTS = "application/vnd.ga4gh.matchmaker.v1.0+json"
 # MME_URL = "http://localhost:9020"
-# MME_TOKEN = "matchmaker_token"
+# MME_TOKEN = "DEMO"
```
Comment (dnil) [inline scout/server/config.py:37]: 😸

File: tests/server/blueprints/cases/test_matchmaker_controllers.py [hunk 5]
```diff
@@ -93,6 +98,61 @@ def test_matchmaker_add(app, user_obj, case_obj, test_hpo_terms, mocker):
...
+        # THEN ine case should be uodated and a submission object correctly saved to database
+        updated_case = store.case_collection.find_one()
+        submission = updated_case["mme_submission"]
+        assert (
+            submission["patients"][0]["genomicFeatures"][0]["variant"]["start"]
+            == test_variant["position"]
+        )
+
+
+def test_matchmaker_add_sv(app, user_obj, case_obj, sv_variant_obj, mocker):
```
Comment (dnil) [inline tests/server/blueprints/cases/test_matchmaker_controllers.py:119]: 🥇

File: tests/server/blueprints/cases/test_matchmaker_controllers.py [hunk 6]
```diff
@@ -102,7 +162,8 @@ def test_matchmaker_add(app, user_obj, case_obj, test_hpo_terms, mocker):
 
         # THEN ine case should be uodated and a submission object correctly saved to database
         updated_case = store.case_collection.find_one()
-        assert len(updated_case["mme_submission"]["patients"]) == 1
+        submission = updated_case["mme_submission"]
+        assert submission["patients"][0]["genomicFeatures"][0]["gene"]["id"] == "TRAPPC9"
```
Comment (dnil) [inline tests/server/blueprints/cases/test_matchmaker_controllers.py:166]: This works ofcourse, but you could make that gene symbol a local constant (variable 😝) if nothing else to speed up reading. I guess fixture is over the top for single use.

**Review by dnil — APPROVED** [review]

**Comment by dnil** [comment]
Nicely done!

File: tests/server/blueprints/cases/test_matchmaker_controllers.py (hunk 6 above)
Comment (northwestwitch) [reply tests/server/blueprints/cases/test_matchmaker_controllers.py]: This is fixed now, I'm going to merge!

**Pull request merged**